V2WebSocketProtocolGuide.md

```diff
@@ -138,23 +138,51 @@ Local proxy will initiate a web socket handshake to connect to the tunnel, using
--   The local proxy, and library clients may use stream ID to determine how to respond to or filter incoming messages
+- The service may use the Service ID to decide how to route traffic between connected tunnel clients. 
+      - For example,  when local proxy received a data packet with Service ID  SSH1, it will look up the configuration for SSH1 and see which port this service ID is mapped to. If SSH1 is mapped to port 22 on local host, then this data packet will be forward to port 22 on local host. 
+- A stream start message may include one of the service ID's defined as part of the tunnel, or no service ID. It may not, however, include any other service ID not defined during tunnel creation.
+- Any subsequent data messages must include a service ID associated with an active or previously active stream (a stream start message for the specific service ID must be sent first). Otherwise, the service will disconnect the client from the tunnel.
+- The local proxy uses the service ID -> stream ID mapping to check the current active stream ID for a specific service ID. 
+- The stream ID validation for a certain stream(service ID) will only be performed on message type _StreamReset_ and _Data_. If a received message failed the stream ID validation, this message is considered to be stale and will be discarded by local proxy. 
+- The local proxy, and library clients may use stream ID to determine how to respond to or filter incoming messages
     -   For example: if a source sends a _StreamStart_ with a stream ID of 345 in response to a newly accepted TCP connection, and afterwards receives a _Data_ message marked with stream ID of 565, that data should be ignored. It's origin is tied to a prior connection over the tunnel from the perspective of the tunnel peer that originated it
     -   Another example: if a source local proxy sends a _StreamStart_ with a stream ID of 345 in response to a newly accepted TCP connection, and afterwards receives a _StreamReset_ message marked with stream ID of 565, that message should be ignored. Only a _StreamReset_ with a stream ID of 345 should cause the client to close its local connection
--   Ending a data stream (normally or abnormally) is accomplished by either side sending a _StreamReset_ with the stream ID that is meant to be closed
--   Locally detected network failures are communicated by sending _StreamReset_ over the tunnel using the active stream ID if one is active.
--   If there is a network issue with the WebSocket connection, no control message is necessary to send. However, the active stream should be considered invalid and closed. Reconnect to the tunnel via the service and start a new stream.
+- Ending a data stream (normally or abnormally) is accomplished by either side sending a _StreamReset_ with the stream ID that is meant to be closed
+- Locally detected network failures are communicated by sending _StreamReset_ over the tunnel using the active stream ID if one is active.
```

Comment:
this should be service id AND stream id since each service id has its own stream id. Sending just a stream id could be confusing as to which specific stream should be closed in the event there are multiple streams going on. Or we can simply state that any network failures should communicate sending the Stream Reset over all active streams (which covers all service/stream ids)

Reply:
I believe the wording of "locally detected network failures" means a TCP socket error within the context of the client application, I adjusted the wording to fit accordingly.